src/nvim/os/env.c

```diff
@@ -613,6 +613,12 @@ char *vim_getenv(const char *name)
     return xstrdup(kos_env_path);
   }
 
+#ifdef WIN32
+  if (STRCMP(name, "HOME") == 0) {
+    return homedir;
```

Comment:
I restored the behavior to match vim (replace `WIN32` with `MSWIN`). I don't know how the `let $HOME = '%USERPROFILE%'` test passed.